sphinx/source/docs/user_guide/server.rst

```diff
@@ -642,6 +642,32 @@ and the server, there is network traffic between the python client and the
 server as well. Depending on the particular usage, this could be a
 significant consideration.
 
+
+Embedding Bokeh Server as a Library
+-----------------------------------
+
+If you want to embed the Bokeh Server in a larger Tornado-based application
+with other concurrent network services or coroutines, here is how to do it:
+
+.. code-block:: python
+
+   def add_bokeh_server(loop, bokeh_applications, **server_kwargs):
+       """
+       Start a Bokeh Server instance in the Tornado *loop*
+       with the given *bokeh_applications*.
+
+       *server_kwargs* can contain additional Bokeh Server arguments,
+       such as the ``port`` number to listen on.
+       """
+       from bokeh.server.server import Server
+       server = Server(bokeh_applications, io_loop=loop,
+                       **server_kwargs)
+       server.start()
+
+When you ever want to stop the embedded Bokeh Server, call the ``stop()``
+method on the given ``Server`` instance.
```

Comment:
My only comment: is it worth explicitly mentioning here that `server.start` does *not* start the passed-in `IOLoop`? It might just be my inexperience with Tornado, but I would expect a method `start` to "start" things, whereas at least it seems to me it's really doing more like a setup step. Or maybe this is a recognized pattern, I'm just not familiar with?